### PR TITLE
Issue 4985: Additional Logs for tracking controller periodic retention 

### LINKED ISSUE #4985: Add more logs to Controller be able to debug execution of periodic retention
**Problem description**
Currently controller logs do not indicate if periodic retention was invoked and weather it had any impact for a given stream. These logs need to be added to controller code so we can track when and how retention was invoked by Controller and what effect it had on the storage consumption of a stream

**Problem location**
https://github.com/pravega/pravega/blob/6acd317344424bbc92c3f870d394f68d93ce1085/controller/src/main/java/io/pravega/controller/task/Stream/StreamMetadataTasks.java#L324

**Suggestions for an improvement**
Add new logs for retention on Controller.

### PR DESCRIPTION
**Pull request by pbelgundi**
**Change log description**  
Added info logs to track execution of controller's periodic retention.

**Purpose of the change**  
Fixes #4985

**What the code does**  
New log messages added

**How to verify it**  
Logs should be noticed when periodic retention is invoked or cannot be invoked because truncation record is not found.

### CODE REVIEW

File: controller/src/main/java/io/pravega/controller/task/Stream/StreamMetadataTasks.java [hunk 1]
```diff
@@ -321,27 +321,32 @@ public void initializeStreamWriters(final EventStreamClientFactory clientFactory
...
                                              RetentionSet retentionSet, StreamCutRecord newRecord, long recordingTime, long requestId) {
-        return findTruncationRecord(policy, retentionSet, newRecord, recordingTime)
-                .map(record -> streamMetadataStore.getStreamCutRecord(scope, stream, record, context, executor)
-                        .thenCompose(streamCutRecord -> startTruncation(scope, stream, streamCutRecord.getStreamCut(), context, requestId))
-                        .thenCompose(started -> {
-                            if (started) {
-                                return streamMetadataStore.deleteStreamCutBefore(scope, stream, record, context, executor);
+        Optional<StreamCutReferenceRecord> truncationRecord = findTruncationRecord(policy, retentionSet, newRecord, recordingTime);
+        if (!truncationRecord.isPresent()) {
+            log.info("Could not find truncation record for stream {}/{}", scope, stream);
```
Comment (shiveshr) [inline controller/src/main/java/io/pravega/controller/task/Stream/StreamMetadataTasks.java:326]: nit: Could not find truncation record seems negative. 
I would reword it to something along the lines of - `No suitable truncation record was found that satisfied retention policy`.

Comment (pbelgundi) [reply controller/src/main/java/io/pravega/controller/task/Stream/StreamMetadataTasks.java]: Ok.
